hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerHA.java

```diff
@@ -759,6 +766,43 @@ public void testReadRequest() throws Exception {
     }
   }
 
+  @Test
+  public void testAddBucketAcl() throws Exception {
+    OzoneBucket ozoneBucket = setupBucket();
+    String remoteUserName = "remoteUser";
+    OzoneAcl defaultUserAcl = new OzoneAcl(USER, remoteUserName,
+        READ, DEFAULT);
+
+    OzoneObj ozoneObj = OzoneObjInfo.Builder.newBuilder()
+        .setResType(OzoneObj.ResourceType.BUCKET)
+        .setStoreType(OzoneObj.StoreType.OZONE)
+        .setVolumeName(ozoneBucket.getVolumeName())
+        .setBucketName(ozoneBucket.getName()).build();
+
+    boolean addAcl = objectStore.addAcl(ozoneObj, defaultUserAcl);
+    Assert.assertTrue(addAcl);
+
+    ozoneBucket.addAcls(Collections.singletonList(defaultUserAcl));
+    List<OzoneAcl> acls = ozoneBucket.getAcls();
+
+    Assert.assertTrue(containsAcl(defaultUserAcl, acls));
+
+    addAcl = objectStore.addAcl(ozoneObj, defaultUserAcl);
+    Assert.assertFalse(addAcl);
+  }
+
+  private boolean containsAcl(OzoneAcl ozoneAcl, List<OzoneAcl> ozoneAcls) {
+    for (OzoneAcl acl : ozoneAcls) {
+      if (acl.getType().equals(ozoneAcl.getType())
+          && acl.getName().equals(ozoneAcl.getName())
+          && acl.getAclBitSet().equals(ozoneAcl.getAclBitSet())
```

Comment:
containersAcl() may not work for the case when the added is OR-ed into existing BitSet where .equals will not be the same?

Reply:
Done.